### PR TITLE
fix(types): Function PropType (#208)

### LINKED ISSUE #208: Function PropType
Hello.

This [Vue issue](https://github.com/vuejs/vue/issues/9692) and this [commit](https://github.com/vuejs/vue/commit/df4af4bd1906b9f23b62816142fdfbd6336d3d2f) refer to allowing:
```
props: {
  fn: Function as PropType<() => any>
}
```
Would it be possible to replicate this change on componentProps.d.ts's PropConstructor type please?

Thanks for your work :)

### PR DESCRIPTION
**Pull request by genie-youn**
Closes #208 

---

Hi~! 😄 
This PR allow using functions on the PropTypes like vue 2.x

Thanks you.

### CODE REVIEW

**Comment by genie-youn** [comment]
ping @posva 😄

File: test/types/createComponent.spec.ts [hunk 1]
```diff
@@ -1,11 +1,9 @@
-import { createComponent, createElement as h, ref, SetupContext } from '../../src';
+import { createComponent, createElement as h, ref, SetupContext, PropType } from '../../src';
 import Router from 'vue-router';
 
 const Vue = require('vue/dist/vue.common.js');
 
-type Equal<Left, Right> = (<U>() => U extends Left ? 1 : 0) extends (<U>() => U extends Right
-  ? 1
-  : 0)
+type Equal<Left, Right> = (<U>() => U extends Left ? 1 : 0) extends <U>() => U extends Right ? 1 : 0
```
Comment (sodatea) [inline test/types/createComponent.spec.ts]: Would you please revert this style change? IMHO the code is more readable with parentheses.

Comment (genie-youn) [reply test/types/createComponent.spec.ts]: @sodatea 

Sure 😄

**Comment by sodatea** [comment]
This branch has conflicts that must be resolved - In the latest codebase, we've moved from `createComponent` to `defineComponent`.

**Comment by genie-youn** [comment]
> This branch has conflicts that must be resolved - In the latest codebase, we've moved from `createComponent` to `defineComponent`.

Okay. I resolved conflicts 😭

File: test/types/defineComponent.spec.ts [hunk 2]
```diff
@@ -85,6 +85,26 @@ describe('defineComponent', () => {
     expect.assertions(3);
   });
 
+  it('custom props type function', () => {
+    interface IPropsTypeFunction {
+      fn: (arg: boolean) => void;
+    }
+    const App = createComponent<IPropsTypeFunction>({
```
Comment (sodatea) [inline test/types/defineComponent.spec.ts]:
```suggestion
    const App = defineComponent<IPropsTypeFunction>({
```

File: test/types/defineComponent.spec.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { createComponent, defineComponent, createElement as h, ref, SetupContext } from '../../src';
+import { createComponent, defineComponent, createElement as h, ref, SetupContext, PropType } from '../../src';
```
Comment (sodatea) [inline test/types/defineComponent.spec.ts]:
```suggestion
import { defineComponent, createElement as h, ref, SetupContext, PropType } from '../../src';
```

**Comment by sodatea** [comment]
❤️Thanks for the help!

**Pull request merged**